Building error: -Wno-class-memaccess, using g++ 9.3.0 on Ubuntu 20.04

I was able to successfully build if I went through Recastdemo.make and Detour.make and removed all the references to -Wno-class-memaccess, but I thought I should post this here in case there's a better way to build it without messing with those files, or in case this is a bug.

```
==== Building DebugUtils (debug) ====
Creating obj/Debug/DebugUtils
DebugDraw.cpp
DetourDebugDraw.cpp
RecastDebugDraw.cpp
RecastDump.cpp
Creating lib/Debug
Linking DebugUtils
==== Building Detour (debug) ====
Creating obj/Debug/Detour
DetourAlloc.cpp
DetourAssert.cpp
DetourCommon.cpp
DetourNavMesh.cpp
DetourNavMeshBuilder.cpp
DetourNavMeshQuery.cpp
DetourNode.cpp
Linking Detour
==== Building DetourCrowd (debug) ====
Creating obj/Debug/DetourCrowd
DetourCrowd.cpp
DetourLocalBoundary.cpp
DetourObstacleAvoidance.cpp
DetourPathCorridor.cpp
DetourPathQueue.cpp
DetourProximityGrid.cpp
Linking DetourCrowd
==== Building DetourTileCache (debug) ====
Creating obj/Debug/DetourTileCache
DetourTileCache.cpp
DetourTileCacheBuilder.cpp
Linking DetourTileCache
==== Building Recast (debug) ====
Creating obj/Debug/Recast
Recast.cpp
RecastAlloc.cpp
RecastArea.cpp
RecastAssert.cpp
RecastContour.cpp
RecastFilter.cpp
RecastLayers.cpp
RecastMesh.cpp
RecastMeshDetail.cpp
RecastRasterization.cpp
RecastRegion.cpp
Linking Recast
==== Building RecastDemo (debug) ====
Creating obj/Debug/RecastDemo
fastlz.c
cc1: error: command line option ‘-Wno-class-memaccess’ is valid for C++/ObjC++ but not for C [-Werror]
cc1: all warnings being treated as errors
make[1]: *** [RecastDemo.make:150: obj/Debug/RecastDemo/fastlz.o] Error 1
make: *** [Makefile:69: RecastDemo] Error 2

```

--- a/RecastDemo/premake5.lua
+++ b/RecastDemo/premake5.lua
@@ -76,7 +76,7 @@ project "Detour"
 	-- linux library cflags and libs
 	filter {"system:linux", "action:gmake"}
 		buildoptions {
-			"-Wno-class-memaccess"
+			"-Wno-error=class-memaccess"
 		}
 
 project "DetourCrowd"
@@ -155,7 +155,7 @@ project "RecastDemo"
 			"`pkg-config --cflags gl`",
 			"`pkg-config --cflags glu`",
 			"-Wno-ignored-qualifiers",
-			"-Wno-class-memaccess"
+			"-Wno-error=class-memaccess"
 
 		}
 		linkoptions { 

